str_test.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env nash
+
+# testing forr str.sh
+
+import ./str
+
+fn testjoin(expected, lst, sep) {
+	got <= join($lst, $sep)
+
+	if $got != $expected {
+		printf "Failed to join: '%s' != '%s'" $got $expected
+
+		return "1"
+	}
+
+	return "0"
+}
+
+# testJoin returns 0 in case of tests pass and 1 otherwise
+fn TestJoin() {
+	tests = (
+		(
+			"scratching my itches"
+			(scratching my itches)
+			" "
+		)
+		(
+			"plan9/from/outer/space"
+			(plan9 from outer space)
+			"/"
+		)
+		(
+			"writing tests in shellscript we are making history"
+			(writing tests in shellscript we are making history)
+			" "
+		)
+	)
```

Comment:
After I thought about it, I don't think this kind of test makes sense anymore. :-)

The `join` function doesn't fail if properly called. If user don't pass a list in `lst` then the interpreter will catch the mistake (when trying to index a non-list variable). The language is dynamic and I don't think it's useful testing for programmers mistake.